pandas/_libs/tslibs/parsing.pyx

```diff
@@ -825,6 +825,10 @@ def format_is_iso(f: str) -> bint:
     iso_template = '%Y{date_sep}%m{date_sep}%d{time_sep}%H:%M:%S.%f'.format
     excluded_formats = ['%Y%m%d', '%Y%m', '%Y']
 
+    if (f is not None) and (f[-2:] in ["SZ", "fZ"]):
```

Comment:
As mentioned in my other comment, shouldn't `%z` be checked in the `iso_template`?

Reply:
I agree, I will revise my PR.